VisualPinball.Engine/VPT/Table/Table.cs

```diff
@@ -177,11 +175,18 @@ public Material GetMaterial(string name)
 				: Data.Materials.FirstOrDefault(m => m.Name == name);
 		}
 
+		public void SetTextureContainer(ITableResourceContainer<Texture> container)
+		{
+			Textures = container;
+		}
+
 		public Texture GetTexture(string name)
 		{
-			return name == null
+			string lowerName = name.ToLower();
+			var tex = name == null
 				? null
-				: Textures.ContainsKey(name.ToLower()) ? Textures[name.ToLower()] : null;

```

Comment:
Can `Textures` *not* contain the name? If not, I prefer a fast fail with a descriptive exception instead of an NPE somewhere later.

Reply:
Sorry, misread the diff, that's actually what you're doing now.